libcontainer/dmz/cloned_binary_linux.go

```diff
@@ -0,0 +1,243 @@
+//go:build linux
+// +build linux
+
+package dmz
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+
+	"github.com/opencontainers/runc/libcontainer/system"
+)
+
+type sealFunc func(**os.File) error
+
+var (
+	_ sealFunc = sealMemfd
+	_ sealFunc = sealFile
+)
+
+func isExecutable(f *os.File) bool {
+	if err := unix.Faccessat2(int(f.Fd()), "", unix.X_OK, unix.AT_EACCESS|unix.AT_EMPTY_PATH); err == nil {
+		return true
+	} else if errors.Is(err, unix.EACCES) {
+		return false
+	}
+	if err := unix.Faccessat(int(f.Fd()), "", unix.X_OK, unix.AT_EACCESS|unix.AT_EMPTY_PATH); err == nil {
+		return true
+	} else if errors.Is(err, unix.EACCES) {
+		return false
+	}
+	path := "/proc/self/fd/" + strconv.Itoa(int(f.Fd()))
+	if err := unix.Access(path, unix.X_OK); err == nil {
+		return true
```

Comment:
The binary might be still unexecutable depending on LSM?
Maybe we should just execute the binary and see whether it returns a magic ret code?

Reply:
We could, though doing a maximum of ~10 extra fork+execs in the absolute worst case seems a bit expensive... This also can't detect every LSM issue (it's possible executing a host `/tmp` program will be blocked by the container's LSM setup -- and we cannot detect that this early into container setup).

Reply:
I think the new `RUNC_DMZ=legacy` option means that users can work around this if they hit an issue here. I don't think there's a nicer way of dealing with this problem completely tbh.